web/src/features/fireWatch/components/fireWatchDashboard.test.tsx

```diff
@@ -0,0 +1,236 @@
+import FireWatchDashboard from '@/features/fireWatch/components/FireWatchDashboard'
+import { FireWatchDetailsModalProps } from '@/features/fireWatch/components/FireWatchDetailsModal'
+import { BurnStatusEnum, FireWatch, FuelTypeEnum, PrescriptionEnum } from '@/features/fireWatch/interfaces'
+import burnForecastSlice, { BurnForecastsState, initialState } from '@/features/fireWatch/slices/burnForecastSlice'
+import { MUI_LICENSE } from '@/utils/env'
+import { LicenseInfo } from '@mui/x-data-grid-pro'
+import { combineReducers, configureStore } from '@reduxjs/toolkit'
+import { act, fireEvent, render, screen, waitFor } from '@testing-library/react'
+import { DateTime } from 'luxon'
+import { Provider } from 'react-redux'
+import { beforeEach, describe, expect, it, vi } from 'vitest'
+
+const buildTestStore = (initialState: BurnForecastsState) => {
+  const rootReducer = combineReducers({ burnForecasts: burnForecastSlice })
+  const testStore = configureStore({
+    reducer: rootReducer,
+    preloadedState: {
+      burnForecasts: initialState
+    }
+  })
+  return testStore
+}
+
+vi.mock('features/fireWatch/fireWatchApi', () => ({
+  getBurnForecasts: () => mockFireWatchBurnForecasts
+}))
+
+vi.mock('@/features/fireWatch/components/FireWatchDetailsModal', () => {
+  return {
+    default: ({ open }: FireWatchDetailsModalProps) => (open ? <div>FireWatch Modal Open</div> : <div></div>)
+  }
+})
+
+vi.mock('@/features/fireWatch/components/DetailPanelContent', () => ({
+  __esModule: true,
+  default: ({ row }: any) => <div data-testid="detail-panel">Detail: {row.title}</div>
+}))
+
+describe('FireWatchDashboard', async () => {
+  beforeEach(() => LicenseInfo.setLicenseKey(MUI_LICENSE))
+
+  it('dispatches fetchBurnForecasts on mount', () => {
+    const testStore = buildTestStore({
+      ...initialState
+    })
+    const dispatchSpy = vi.spyOn(testStore, 'dispatch')
+    render(
+      <Provider store={testStore}>
+        <FireWatchDashboard />
+      </Provider>
+    )
+    expect(dispatchSpy).toHaveBeenCalled()
+  })
+
+  it('should render', async () => {
+    const testStore = buildTestStore({
+      ...initialState
+    })
+    await act(async () =>
```

Comment:
Is `act` an alternative to `waitFor`?

Reply:
They are somewhat similar but there seems to be some nuance. Typically you wouldn't even need to use `act` in conjunction with the `render` call, but I believe it is necessary in this case when waiting on a `dispatch` call that eventually updates state. I don't think `act` will return until those state updates are complete. In comparison, `waitFor` implements a poling mechanism and repeatedly checks to see if the `expect` returns `true` (in this use case) and times out after 1000ms.

Reply:
Interesting, where did you find those details? Looking at https://testing-library.com/docs/react-testing-library/api/#act, which circularly references: https://react.dev/reference/react/act, it looks like react-testing library wraps the behaviour of act in it's helper functions already. Does the test fail with `waitFor`?

Reply:
Tests seem to pass with `waitFor`. I assume this is due to the delay it introduces versus actually waiting for rendering to happen though. `waitFor` waits for the callback function's promise to resolve. As `render` itself doesn't actually return a promise, the `async () => render()` likely returns after the first 50ms second delay introduced by `waitFor`. I believe `waitFor` actually comes from `@testing-library/dom-testing-library` and does not include a call to `act`. I'll admit I don't fully understand all the nuances. Error messages from either `react-testing-library` or `dom-testing-library` told me to use `act`!

Reply:
Interesting, yeah let's stick with `act` if there's no async promise that's being awaited.